pkg/config/mesh/mesh.go

```diff
@@ -122,27 +123,46 @@ func ApplyProxyConfig(yaml string, meshConfig meshconfig.MeshConfig) (*meshconfi
 	return mc, nil
 }
 
+func extractProxyConfig(yamlText string) (string, error) {
+	mp := map[string]interface{}{}
+	if err := yaml.Unmarshal([]byte(yamlText), &mp); err != nil {
+		return "", err
+	}
+	proxyConfig := mp["defaultConfig"]
+	if proxyConfig == nil {
+		return "", nil
+	}
+	bytes, err := yaml.Marshal(proxyConfig)
+	if err != nil {
+		return "", err
+	}
+	return string(bytes), nil
+}
+
 // ApplyMeshConfig returns a new MeshConfig decoded from the
 // input YAML with the provided defaults applied to omitted configuration values.
 func ApplyMeshConfig(yaml string, defaultConfig meshconfig.MeshConfig) (*meshconfig.MeshConfig, error) {
-	if err := gogoprotomarshal.ApplyYAML(yaml, &defaultConfig); err != nil {
-		return nil, multierror.Prefix(err, "failed to convert to proto.")
-	}
+	// We want to keep semantics that all fields are overrides, except proxy config is a merge. This allows
```

Comment:
note: I am not changing the semantics here, just documenting them. The only thing I have changed is allowing overrides with default (as defined by go zero values) values